exchanges/poloniex/poloniex_futures.go

```diff
@@ -0,0 +1,729 @@
+func (e *Exchange) GetFuturesLeverage(ctx context.Context, symbol, marginMode string) ([]*FuturesLeverage, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	if marginMode != "" {
+		params.Set("mgnMode", marginMode)
+	}
+	var resp []*FuturesLeverage
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodGet, positionPathV3+"leverages", params, nil, &resp)
+}
+
+// SetFuturesLeverage change leverage
+func (e *Exchange) SetFuturesLeverage(ctx context.Context, symbol, marginMode, positionSide string, leverage int64) (*FuturesLeverage, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	if marginMode == "" {
+		return nil, margin.ErrInvalidMarginType
+	}
+	if positionSide == "" {
+		return nil, order.ErrSideIsInvalid
+	}
+	if leverage <= 0 {
+		return nil, order.ErrSubmitLeverageNotSupported
+	}
+	var resp *FuturesLeverage
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodPost, positionPathV3+"leverage", nil, &map[string]string{
+		"symbol":  symbol,
+		"mgnMode": marginMode,
+		"posSide": positionSide,
+		"lever":   strconv.FormatInt(leverage, 10),
+	}, &resp)
+}
+
+// SwitchPositionMode switch the current position mode. Please ensure you do not have open positions and open orders under this mode before the switch.
+// Position mode, HEDGE: LONG/SHORT, ONE_WAY: BOTH
+func (e *Exchange) SwitchPositionMode(ctx context.Context, positionMode string) error {
+	if positionMode == "" {
+		return errPositionModeInvalid
+	}
+	return e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodPost, positionPathV3+"mode", nil, map[string]string{"posMode": positionMode}, &struct{}{})
+}
+
+// GetPositionMode get the current position mode.
+func (e *Exchange) GetPositionMode(ctx context.Context) (string, error) {
+	resp := &struct {
+		PositionMode string `json:"posMode"`
+	}{}
+	return resp.PositionMode, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodGet, positionPathV3+"mode", nil, nil, &resp)
+}
+
+// GetFuturesOrderBook get market depth data of the designated trading pair
+func (e *Exchange) GetFuturesOrderBook(ctx context.Context, symbol string, depth, limit int64) (*FuturesOrderbook, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	if depth > 0 {
+		params.Set("scale", strconv.FormatInt(depth, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp *FuturesOrderbook
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"orderBook", params), &resp)
+}
+
+// IntervalToString returns a string representation of kline.Interval instance.
+func IntervalToString(interval kline.Interval) (string, error) {
```

Comment:
🚧 Change request:
Note: I also asked [here](https://github.com/thrasher-corp/gocryptotrader/pull/1351#discussion_r2390216650) for this to be moved to `poloniex.go` next to the type and the other method 🙂 
And I've just spotted that this is exported, and I don't think it should be
```suggestion
// intervalToString returns a string representation of kline.Interval instance.
func intervalToString(interval kline.Interval) (string, error) {
    panic("Please don't just accept the suggestion. Also move me to next to my family. I miss them")
```